### PR TITLE
chore(e2e) Add E2E tests for export COMPASS-5467

### PR DESCRIPTION
**Pull request by lerouxb**
One of them is currently skipped because exporting to JSON doesn't currently do what you ask for.

### CODE REVIEW

File: packages/compass-e2e-tests/tests/collection-export.test.ts [hunk 1]
```diff
@@ -135,11 +141,324 @@ describe('Collection export', function () {
...
+      all_fields: true,
+      number_of_docs: 1000,
+      success: true,
+    });
+    expect(telemetry.screens()).to.include('export_modal');
+  });
+
+  // Skipping for now because export to JSON seems to include the _id field even
+  // if we explicitly deselect it.
+  it.skip('supports full collection to JSON with a subset of fields', async function () {
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/collection-export.test.ts:409]: https://jira.mongodb.org/browse/COMPASS-5509

File: packages/compass-e2e-tests/tests/collection-export.test.ts [hunk 2]
```diff
@@ -135,11 +141,324 @@ describe('Collection export', function () {
...
-  it('supports collection to CSV with a subset of fields');
-  it('supports collection to JSON with a query filter');
-  it('supports full collection to JSON');
-  it('supports collection to JSON with all fields');
-  it('supports collection to JSON with a subset of fields');
+  it('supports full collection to CSV with all fields', async function () {
+    const telemetryEntry = await browser.listenForTelemetryEvents(telemetry);
+
+    // we're not going to use this query, but we're running it so we can make
+    // sure it gets ignored
```
Comment (addaleax) [inline packages/compass-e2e-tests/tests/collection-export.test.ts:148]: Just for my own understanding, this is “ignored” as in “not used as a filter for exported data”, not “ignored”  as in “not displayed in the export modal”, right?

**Review by addaleax — APPROVED** [review]

File: packages/compass-e2e-tests/tests/collection-export.test.ts (hunk 2 above)
Comment (lerouxb) [reply packages/compass-e2e-tests/tests/collection-export.test.ts:148]: Gets displayed (few lines lower I think), but not used becuase we're selecting the other radio button.

Comment (lerouxb) [reply packages/compass-e2e-tests/tests/collection-export.test.ts:148]: ie. That matches just one doc but we're going to make sure all 1000 got exported.

**Comment by lerouxb** [comment]
Windows got a flake where hovering over the new collection in the sidebar never showed the actions. I'll have to just try and get .hover() to be more robust in a separate PR. That's not new code anyway.

**Pull request merged**